test/controllers/api/users_controller_test.rb

```diff
@@ -487,6 +502,26 @@ def check_xml_details(user, include_private, include_email)
     def check_json_details(js, user, include_private, include_email)
       assert_equal user.id, js["user"]["id"]
       assert_equal user.description, js["user"]["description"]
+      if user.company
+        assert_equal user.company, js["user"]["company"]
+      else
+        assert_nil js["user"]["company"]
+      end
+
+      if user.social_links.present?
+        js_links = js.dig("user", "social_links")
+        assert_equal user.social_links.count, js_links.count
+        user.social_links.each do |user_link|
+          details = user_link.parsed
+          matching_js_link = js_links.find { |js_link| 
+            js_link["url"] == details[:url] && js_link["platform"] == details[:platform] 
+          }
+          assert_not_nil matching_js_link
+        end
```

Comment:
Alternatively, you can sort them first and avoid the nested loop:

```suggestion
        ordered_js_links = js.dig("user", "social_links").sort_by { |link| link["url"] }
        ordered_user_links = user.social_links.sort_by(&:url).map(&:parsed)

        assert_equal ordered_user_links.count, ordered_js_links.count

        ordered_user_links.zip(ordered_js_links).each do |user_link, js_link|
          assert_equal user_link[:url], js_link["url"]
          assert_equal user_link[:platform], js_link["platform"]
        end
```